### PR TITLE
Fix `db:prepare` task to create/migrate test db when dev db already exists

### LINKED ISSUE #44815: db:prepare task fails to create/migrate test db when dev db already exists
When a dev db already exists, the test db is not created or setup by `db:prepare` task nor the `bin/setup` script that runs it.

We have our dev db provided (with data) by a docker image. The test db is stood up by the app. However, due to this issue, `bin/setup` nor `db:prepare` will correctly create the test db because the dev db already exists.

Given that `db:setup` _does_ properly create the test db (even when dev db exists), it seems reasonable that `db:prepare` would be similarly idempotent and result in a stable dev&test setup.

### Steps to reproduce

In a brand new rails app (`rails new testcase --api --database mysql`)

1. prepare databases
```sh
$ bin/rails db:prepare
Created database 'testcase_development'
Created database 'testcase_test'
```
2. Drop the test db leaving development db only. 
```sh
$ RAILS_ENV=test rake db:drop
Dropped database 'testcase_test'
```
3. Re-run db:prepare
```sh
$ bin/rails db:prepare
```

### Expected behavior

I would expect the test db to be created/seeded/migrated.
The `db:prepare` task (and the `bin/setup` in which it is used) both document that the task should be idempotent and that it will set up dev and test databases.
If the dev db already exists, it should continue on to migrate or seed (as appropriate) and then do a full setup by creating and seeding a test database as well.

Expected output:
```sh
$ bin/rails db:prepare
Database 'testcase_development' already exists
Created database 'testcase_test'
```

(For the record, this _is_ the actual behavior for `db:setup`; if the dev db already exists, it moves on to create+seed the test db.)

### Actual behavior

Nothing happens. The dev db is detected, and then the test db preparation is skipped entirely.

### System configuration
**Rails version**:
```
$ rails --version
Rails 7.0.2.3
```
**Ruby version**:
```
$ ruby --version
ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [x86_64-darwin21]
```

### PR DESCRIPTION
**Pull request by fatkodima**
Fixes #44815.

### CODE REVIEW

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+*   Fix `db:prepare` task to create/migrate test db when dev db already exists.
+
+    Fixes #44815.
+
+    *fatkodima*
```
Comment (eileencodes) [inline activerecord/CHANGELOG.md:5]: Can you remove the changelog entry. We don't usually mention bug fixes.

**Review by eileencodes — COMMENTED** [review]
This looks great, thanks for the PR! Let me know when you remove the changelog and I'll merge.

File: activerecord/CHANGELOG.md (hunk 1 above)
Comment (fatkodima) [reply activerecord/CHANGELOG.md:5]: Sure, but I'm seeing several bug fixes in the changelog - https://github.com/rails/rails/blob/main/activerecord/CHANGELOG.md ?

Comment (eileencodes) [reply activerecord/CHANGELOG.md:5]: Well, they weren't supposed to be added so let's not add one more that will need to be removed.

Comment (fatkodima) [reply activerecord/CHANGELOG.md]: 👍 Removed.

**Comment by eileencodes** [comment]
Thanks for the fix!

**Comment by fatkodima** [comment]
@eileencodes `db:prepare` was introduced in 6.0 - https://github.com/rails/rails/pull/35768 
Tested for 6.0 and 6.1 - it was broken since then.

**Comment by eileencodes** [comment]
I just remembered that we are only doing bug fix backports to 7.0 so I wont be backporting this to 6.0 or 6.1.

**Comment by jasonkarns** [comment]
Thanks for the fix! and holy moly so fast! 🎉

**Pull request merged**